src/math.rs

```diff
@@ -0,0 +1,68 @@
+// Manually translated to Rust from FastLED's MIT licensed C++ code
+
+/// Scale a 16-bit unsigned value by an 8-bit value, which is treated
+/// as the numerator of a fraction whose denominator is `u8::MAX`.
+///
+/// In other words, it computes `i * (scale / u8::MAX)`
+pub(crate) fn scale16by8(i: u16, scale: u8) -> u16 {
+    if scale == 0 {
+        return 0;
+    }
+    ((i as u32 * (1 + scale as u32)) >> 8) as u16
+}
+
+/// Maps an integer from one integer size to another.
+///
+/// For example, a value representing 40% as a `u16` would be `26,214 / 65,535`.
+/// Converting that to a `u8` would return `102 / 255`, exactly 40% through the
+/// range of values representable by a `u8`.
+pub(crate) fn map16_to_8(x: u16) -> u8 {
+    // Tested to be nearly identical to double precision floating point
+    // doing this operation.
+    if x == 0 {
+        return 0;
+    }
```

Comment:
Why is this a special case? Does ((0 + 128) >> 8) not result in the same thing?

Reply:
You're right, there's no need for this special case handling 0.